### PR TITLE
feat: Add source deploys and traffic updates

### LINKED ISSUE #16: Consider not requiring `image` and supporting build and deploy from source
I see that deploying to Cloud Run the action requires an `image` attribute. That makes sense if the user wants to use one step to build a container, and then another step deploy it.
```
steps:
- id: deploy
  uses: google-github-actions/deploy-cloudrun@main
  with:
    image: gcr.io/cloudrun/hello
    service: hello-cloud-run
    credentials: ${{ secrets.gcp_credentials }}
```

But we could imagine 
```
steps:
- id: deploy
  uses: google-github-actions/deploy-cloudrun@main
  with:
    service: hello-cloud-run
    credentials: ${{ secrets.gcp_credentials }}
```

Which would build and deploy the current repo.

Under the hood, `gcloud beta run deploy --source .` would be used.

### LINKED ISSUE #39: Custom revision name as action input
### TL;DR
Hi! I want to give my revisisons a customised name from this action, whithout using a service.yaml metadata file.

### Design

**Action YAML**

The main use case here I see is including the commit hash from the Actions runner in the revision name, something like the following:

```yaml
    - name: deploy
      uses: google-github-actions/deploy-cloudrun@main
      with:
        region: ${{ env.REGION }}
        image: gcr.io/cloudrun/hello
        revision: hello-${{ github.sha }}
```

With the resulting revision being resembling `hello-18c8429f46c5f2717aece30ec83cf83d04943fda`

**Resources**
N/A

**Additional information**

There may be a limit on the revision name, and unfortuantely it's not trival to get a short commit hash directly from GitHub, so that could be a blocker here.

### PR DESCRIPTION
**Pull request by averikitsch**
Fixes #16 - input field 'source'
Fixes #39 - input field 'suffix'

### CODE REVIEW

File: .github/workflows/deploy-cloudrun.yml [hunk 1]
```diff
@@ -22,12 +22,5 @@ jobs:
...
-    - uses: google-github-actions/setup-gcloud@master
-      with:
-        service_account_key: ${{ secrets.DEPLOY_CLOUDRUN_SA_KEY_B64 }}
-
     - name: npm test
-      run: npm run test
-      env:
-        TEST_DEPLOY_CLOUDRUN_CREDENTIALS: ${{ secrets.DEPLOY_CLOUDRUN_SA_KEY_B64 }}
-        TEST_DEPLOY_CLOUDRUN_PROJECT: ${{ secrets.DEPLOY_CLOUDRUN_PROJECT_ID }}
+      run: npm run test
```
Comment (bharathkkb) [inline .github/workflows/deploy-cloudrun.yml:26]: nit: newline

File: action.yml [hunk 2]
```diff
@@ -60,6 +60,73 @@ inputs:
       setup-gcloud action with exported credentials.
     required: false
 
+  source:
+    description: |-
+      The location of the source to build. The location can be a directory on
+      a local disk or a gzipped archive file (.tar.gz) in Google Cloud Storage.
```
Comment (bharathkkb) [inline action.yml:66]: Any additional permissions needed when using GCS?

File: action.yml [hunk 3]
```diff
@@ -60,6 +60,73 @@ inputs:
...
+      Comma separated list of traffic assignments in the form TAG=PERCENTAGE.
+      TAG must match a traffic tag on a revision of the service. PERCENTAGE must
+      be an integer percentage between 0 and 100 inclusive. Up to 100 percent of
+      traffic may be assigned. If 100 percent of traffic is assigned, the service
+      traffic is updated as specified. If under 100 percent of traffic is 
+      assigned, the service traffic is updated as specified to the given tags,
+      and other traffic is scaled up or down proportionally.
+    required: false
+
+  flags:
```
Comment (bharathkkb) [inline action.yml:116]: Should maybe some of the other params like `--no-traffic` be condensed into this? 

Also would there be a case where the flag may contain a space?

File: src/deploy-cloudrun.ts [hunk 4]
```diff
@@ -37,12 +37,61 @@ export async function run(): Promise<void> {
...
+        name,
+        '--quiet',
+        '--platform',
+        'managed',
+        '--region',
+        region,
+        '--source',
+        source,
+      ];
+      installBeta = true;
```
Comment (bharathkkb) [inline src/deploy-cloudrun.ts:93]: Should we extract setting the beta in cmd into a common func where we can keep a list of beta commands? This way we can keep track of it in a single place when gcloud promotes to non beta.

File: src/deploy-cloudrun.ts [hunk 5]
```diff
@@ -133,12 +196,12 @@ export async function run(): Promise<void> {
       },
       silent: true,
     };
-
+    core.info(`running: ${toolCommand} ${cmd.join(' ')}`);
```
Comment (bharathkkb) [inline src/deploy-cloudrun.ts:199]: Good idea! This would be very useful for debugging

**Review by bharathkkb — APPROVED** [review]
LGTM, minor comments

File: action.yml (hunk 2 above)
Comment (averikitsch) [reply action.yml:66]: Yes, good catch.

File: action.yml (hunk 3 above)
Comment (averikitsch) [reply action.yml:116]: I like surfacing this set of flags functionality (traffic management, version tagging) explicitly through the inputs. I've updated the flag splitting to not split if in quotes with tests.

File: src/deploy-cloudrun.ts (hunk 4 above)
Comment (averikitsch) [reply src/deploy-cloudrun.ts:93]: I like this idea, but I am not sure how that would be implemented.

Comment (bharathkkb) [reply src/deploy-cloudrun.ts:93]: We could keep a list of beta flags in a list and do a lookup before we exec to check if any cmd args is in beta list.

**Pull request merged**